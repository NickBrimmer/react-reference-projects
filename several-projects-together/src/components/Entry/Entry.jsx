import React from 'react';
import PropTypes from 'prop-types';

const Entry = ({ year, text }) => (
  <section>
    <p>{year}</p>
    <p>{text}</p>
  </section>
);

Entry.propTypes = {
  year: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default Entry;
