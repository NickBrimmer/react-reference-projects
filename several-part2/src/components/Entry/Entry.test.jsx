import React from 'react';
import { shallow } from 'enzyme';
import Entry from './Entry';

describe('Entry component', () => {
  it('renders Entry', () => {
    const wrapper = shallow(<Entry year="2020" text="this app" />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
