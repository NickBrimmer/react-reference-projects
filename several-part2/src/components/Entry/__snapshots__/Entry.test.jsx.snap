// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry component renders Entry 1`] = `
<section>
  <p>
    2020
  </p>
  <p>
    this app
  </p>
</section>
`;
